src/features/ExtensionCommands.ts

```diff
@@ -388,44 +390,106 @@ export class ExtensionCommandsFeature implements IFeature {
+                    newFileAbsolutePath = saveFileDetails.newPath;
+                } else {
+                    // If not, interpret the path as relative to the workspace root
+                    const workspaceRootUri = vscode.workspace.workspaceFolders[0].uri;
+                    // We don't saving to a non-file URI-schemed workspace
+                    if (workspaceRootUri.scheme !== "file") {
+                        return EditorOperationResponse.Completed;
+                    }
+                    newFileAbsolutePath = path.join(workspaceRootUri.fsPath, saveFileDetails.newPath);
+                }
+
+                await this.saveDocumentContentToAbsolutePath(currentFileUri, newFileAbsolutePath);
+                return EditorOperationResponse.Completed;
+
+            default:
+                // Other URI schemes are not supported
+                const msg = JSON.stringify(saveFileDetails);
+                this.log.writeVerbose(
+                    `<${ExtensionCommandsFeature.name}>: Saving a document with scheme '${currentFileUri.scheme}' ` +
+                    `is currently unsupported. Message: '${msg}'`);
+                return EditorOperationResponse.Completed;
         }
+    }
 
-        // Otherwise we want to save as a new file
-
-        // First turn the path we were given into an absolute path
-        // Relative paths are interpreted as relative to the original file
-        const newFileAbsolutePath = path.isAbsolute(saveFileDetails.newPath) ?
-            saveFileDetails.newPath :
-            path.resolve(path.dirname(saveFileDetails.filePath), saveFileDetails.newPath);
-
-        // Retrieve the text out of the current document
-        const oldDocument = await vscode.workspace.openTextDocument(saveFileDetails.filePath);
-
-        // Write it to the new document path
-        fs.writeFileSync(newFileAbsolutePath, oldDocument.getText());
-
-        // Finally open the new document
-        const newFileUri = vscode.Uri.file(newFileAbsolutePath);
-        const newFile = await vscode.workspace.openTextDocument(newFileUri);
-        vscode.window.showTextDocument(newFile, { preview: true });
+    /**
+     * Take a document available to vscode at the given URI and save it to the given absolute path
+     * @param documentUri the URI of the vscode document to save
+     * @param destinationAbsolutePath the absolute path to save the document contents to
+     */
+    private async saveDocumentContentToAbsolutePath(
+        documentUri: vscode.Uri,
+        destinationAbsolutePath: string): Promise<void> {
+            // Retrieve the text out of the current document
+            const oldDocument = await vscode.workspace.openTextDocument(documentUri);
+
+            // Write it to the new document path
+            try {
+                // TODO: Change this to be asyncronous
+                fs.writeFileSync(destinationAbsolutePath, oldDocument.getText());
```

Comment:
Can we not already make this async? If it's too much work don't worry about it.

Synchronously make the file, asynchronously add the content to the file?

Reply:
I looked into it, but `writeFile` is an older node API that wants a callback. I have been contemplating turning the whole call into a `Promise` though. Thoughts?

Reply:
Sorry if this is already known, but FYI you can await inline promises, e.g:

```typescript
await new Promise<void>((resolve) => {
    fs.writeFile(destinationAbsolutePath, oldDocument.getText(), resolve);
});
```

Reply:
Ah that's handy! Thanks @SeeminglyScience

Reply:
Type system wasn't quite happy with `resolve` there, but I've ended up with something similar.